includes/wikia/Wikia.php

```diff
@@ -140,27 +140,26 @@ public static function initAsyncRequest( $wikiId, $user = null ) {
 	}
 
 	public static function getFaviconFullUrl() {
-		global $wgMemc;
+		return WikiaDataAccess::cache(
+			wfMemcKey( self::FAVICON_URL_CACHE_KEY ),
+			WikiaResponse::CACHE_STANDARD,
+			function () {
+				$faviconFilename = 'Favicon.ico';
 
-		$mMemcacheKey = wfMemcKey(self::FAVICON_URL_CACHE_KEY);
-		$mData = $wgMemc->get($mMemcacheKey);
-		$faviconFilename = 'Favicon.ico';
+				$localFaviconTitle = Title::newFromText($faviconFilename, NS_FILE);
 
-		if ( empty($mData) ) {
-			$localFaviconTitle = Title::newFromText( $faviconFilename, NS_FILE );
-			#FIXME: Checking existance of Title in order to use File. #VID-1744
-			if ( $localFaviconTitle->exists() ) {
-				$localFavicon = wfFindFile( $faviconFilename );
-			}
-			if ( $localFavicon ) {
-				$favicon = $localFavicon->getURL();
-			} else {
-				$favicon = GlobalFile::newFromText( $faviconFilename, self::COMMUNITY_WIKI_ID )->getURL();
-			}
-			$wgMemc->set($mMemcacheKey, $favicon, 86400);
-		}
+				#FIXME: Checking existance of Title in order to use File. #VID-1744
```

Comment:
Checking for existence is worth anyway. Maybe even an assertion here so that we have a ticket created automatically when it starts failing.